OpenMEEG/include/sensors.h

```diff
@@ -92,74 +93,85 @@ namespace OpenMEEG {
         void setPosition(size_t idx, Vector& pos); /*!< Set the position (3D point) of the integration point i. */
-        void setOrientation(size_t idx, Vector& orient); /*!< Set the orientation (3D point) of the integration point i. */
 
         bool hasSensor(std::string name) const;
         size_t getSensorIdx(std::string name) const;
-        Triangles getInjectionTriangles(size_t idx) const { om_assert(idx < m_triangles.size()); return m_triangles[idx]; } /*!< For EIT, get triangles under the current injection electrode. */
-
-        Vector getRadii()   const { return m_radii; }
-        Vector getWeights() const { return m_weights; }
-
-        SparseMatrix getWeightsMatrix() const;
-
-        bool isEmpty() { if(m_nb == 0) return true; else return false; } /*!< Return if the sensors object is empty. The sensors object is empty if its number of sensors is null. */
-        void info() const; /*!< \brief get info about sensors. */
+        bool isEmpty() const { return (m_nb == 0); } /*!< Return if the sensors object is empty. The sensors object is empty if its number of sensors is null. */
+        void info(int n_lines = 5) const; /*!< \brief get n_lines first lines info about sensors. */
 
-    private:
-        size_t m_nb;                        /*!< Number of sensors. */
-        Strings m_names;                    /*!< List of sensors names. */
-        Matrix m_positions;                 /*!< Matrix of sensors positions. ex: positions(i,j) with  j in {0,1,2} for sensor i */
-        Matrix m_orientations;              /*!< Matrix of sensors orientations. ex: orientation(i,j) with  j in {0,1,2} for sensor i */
-        Vector m_weights;                   /*!< Weights of integration points */
-        Vector m_radii;                     /*!< Areas of the EIT sensors */
-        std::vector<Triangles> m_triangles; /*!< Triangles under each EIT sensors */
-        const Geometry * m_geo;             /*!< Geometry on which are applied EIT sensors */
+    protected:
+        size_t m_nb;                          /*!< Number of sensors. */
+        Strings m_labels;                     /*!< List of sensors names. */
+        Matrix m_positions;                   /*!< Matrix of sensors positions. ex: positions(i,j) with  j in {0,1,2} for sensor i */
         std::vector<size_t> m_pointSensorIdx; /*!< Correspondance between point id and sensor id */
-        void findInjectionTriangles();      /*!< Get the triangles under each EIT sensors */
     };
 
     inline Vector Sensors::getPosition(size_t idx) const {
         return m_positions.getlin(idx);
     }
 
-    inline Vector Sensors::getOrientation(size_t idx) const {
-        return m_orientations.getlin(idx);
-    }
-
     inline void Sensors::setPosition(size_t idx, Vector& pos) {
         return m_positions.setlin(idx,pos);
     }
+}
 
-    inline void Sensors::setOrientation(size_t idx, Vector& orient) {
-        return m_orientations.setlin(idx,orient);
+namespace {
```

Comment:
I don't think that code should be placed in the header. There's no need to remove `sensors.cpp` and let there whatever belongs there.